Makefile

```diff
@@ -36,16 +37,16 @@ rm-sequential:
 start-server: $(WS_PAR)/mets.xml
 	ocrd workspace -d $(WS_PAR)  -U $(SOCK) server start
 
-rm-parallel:
+stop-server: $(WS_PAR)/mets.xml
+	-ocrd workspace -d $(WS_PAR)  -U $(SOCK) server stop
+
+rm-parallel: stop-server
 	rm -rf $(WS_PAR)
 
 $(WS_PAR)/mets.xml:
 	cp -r $(WS) $(WS_PAR)
 
 parallel: $(WS_PAR)/mets.xml
 	cd $(WS_PAR) ; \
-	page_ranges=( $$(ocrd workspace list-page -D $(NUMBER_OF_THREADS) -f comma-separated) ) ;\
-	for chunk in $$(seq 0 $$(( $(NUMBER_OF_THREADS) -1 )));do \
-		$(PROCESSOR) -U $(SOCK) -I DEFAULT -O VANDALIZED -g $${page_ranges[$$chunk]} &\
-	done; wait
+	ocrd workspace list-page | parallel -j $(NUMBER_OF_THREADS) "timeout $(PAGE_TIMEOUT) $(PROCESSOR) -U $(SOCK) -I DEFAULT -O VANDALIZED -g {} || $(ECHO) ocrd-dummy -U $(SOCK) -I DEFAULT -O VANDALIZED -g {}"
```

Comment:
LGTM, somehow using `parallel` didn't occur to me.

But could we retain the original variant as `make parallel-chunked`, so we still have an example of `ocrd workspace list-page -D .. -C ..`?

We could also assign the `timeout $(PROCESSOR) ... | ocrd-dummy ...` construct to a variable. How should we document this? Or could we even find a clever way to provide this as a processor-independent script in core?

Reply:
> But could we retain the original variant as `make parallel-chunked`, so we still have an example of `ocrd workspace list-page -D .. -C ..`?

done (rebased and force-pushed).

> We could also assign the `timeout $(PROCESSOR) ... | ocrd-dummy ...` construct to a variable. How should we document this? Or could we even find a clever way to provide this as a processor-independent script in core?

Good questions. I will have ponder over this a while.

Reply:
Originally, we planned to implement [error handling in the Processor class](https://github.com/OCR-D/core/issues/322). (So this would apply everywhere, even the servers would have benefitted.) But I now doubt that change will ever come to fruition. With the above workaround recipe, scripts become more complicated, and one has to choose between page-wise (better for error handling) or chunk-wise (more efficient). But it's better than nothing.

Practically, `$(PROCESSOR)` will be a complete pipeline (using `ocrd-make` or `ocrd process` or a sequence of processor calls). Similar things can be done in Nextflow scripts. Only for the network implementation it is difficult to provide something similar.

Reply:
BTW, we could also provide this per-page parallelism recipe in core via Python. For the user, it could then look like

    ocrd process --jobs 4 --timeout 2m --on-error=empty

What do you think, @kba?

Reply:
That sounds very reasonable and would make life easier right away in deployments. Let's open an issue for this in OCR-D/core.

Reply:
To be continued in https://github.com/OCR-D/core/issues/1155